File: user/src/main/java/com/kjj/user/client/ClientUriTool.java
package com.kjj.user.client;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
public class ClientUriTool {

    @Value("${kjj.client.host}") private String clientServerHost;

    public String getRequestURI(String path, Map<String, String> params) {
        StringBuilder stringBuilder = new StringBuilder(clientServerHost + path + '?');
        for (Map.Entry<String, String> entry : params.entrySet()) {
            stringBuilder.append(String.join("=", entry.getKey(), entry.getValue()));
            stringBuilder.append("&");
        }
        return stringBuilder.toString();
    }
}

File: user/src/main/kotlin/com/kjj/user/client/ClientUriTool.kt
package com.kjj.user.client

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class ClientUriTool {
    @Value("\${kjj.client.host}")
    lateinit var clientServerHost: String

    fun getRequestUri(path: String, params: Map<String, String>): String {
        val stringBuilder = StringBuilder("$clientServerHost$path?")
        for ((key, value) in params) {
            stringBuilder.append("$key=$value&")
        }
        return stringBuilder.toString()
    }
}